extensions/azure/processors/FetchAzureBlobStorage.cpp

```diff
@@ -0,0 +1,146 @@
+/**
+ * @file FetchAzureBlobStorage.cpp
+ * FetchAzureBlobStorage class implementation
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "FetchAzureBlobStorage.h"
+
+#include "core/Resource.h"
+#include "utils/ProcessorConfigUtils.h"
+
+namespace org::apache::nifi::minifi::azure::processors {
+
+const core::Property FetchAzureBlobStorage::RangeStart(
+  core::PropertyBuilder::createProperty("Range Start")
+    ->withDescription("The byte position at which to start reading from the blob. An empty value or a value of zero will start reading at the beginning of the blob.")
+    ->supportsExpressionLanguage(true)
+    ->build());
+const core::Property FetchAzureBlobStorage::RangeLength(
+  core::PropertyBuilder::createProperty("Range Length")
+    ->withDescription("The number of bytes to download from the blob, starting from the Range Start. "
+                      "An empty value or a value that extends beyond the end of the blob will read to the end of the blob.")
+    ->supportsExpressionLanguage(true)
+    ->build());
+
+const core::Relationship FetchAzureBlobStorage::Success("success", "All successfully processed FlowFiles are routed to this relationship");
+const core::Relationship FetchAzureBlobStorage::Failure("failure", "Unsuccessful operations will be transferred to the failure relationship");
+
+namespace {
+class WriteCallback : public OutputStreamCallback {
+ public:
+  WriteCallback(storage::AzureBlobStorage& azure_blob_storage, const storage::FetchAzureBlobStorageParameters& params, std::shared_ptr<core::logging::Logger> logger)
+    : azure_blob_storage_(azure_blob_storage),
+      params_(params),
+      logger_(std::move(logger)) {
```

Comment:
The logger is not used, consider removing it.

Reply:
Removed it in 1180ddd237dae38a7f000c250be3ec3cc3c025d4